### PR TITLE
Completed Work

### PR DESCRIPTION
**Pull request by AvigailGellis**
Here are my changes, review the work and comment in the conversation tab of the pull request.

### CODE REVIEW

File: BakeryDB/1 tables.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    Quantity int not null constraint ck_Orders_Quantity_must_be_greater_than_zero check(Quantity >= 0),
+    PricePerUnit decimal(6,2) not null constraint ck_Orders_PricePerUnit_must_be_between_2_dollars_and_50_cents_and_650_dollars check(PricePerUnit between 2.5 and 650),
+    TotalCost as Quantity * PricePerUnit,
+    DateOrdered date not null constraint ck_DateOrdered_must_be_before_the_current_date check(DateOrdered <= getdate()),
+    DateNeeded date not null constraint ck_Orders_DateNeeded_must_be_after_January_1_2023 check(DateNeeded >= '2023-01-01'),
+    CreditCardNum char(4) not null constraint ck_Orders_CreditCardNum_must_be_numbers check(CreditCardNum like ('[0-9][0-9][0-9][0-9]') or CreditCardNum = ''),
+    CommunicationType varchar(20) not null constraint ck_Orders_CommnicationType_must_be_call_text_mail_or_email check(CommunicationType in ('Call', 'Text', 'Mail', 'Email')),
+    PaymentType char(4) not null constraint ck_Orders_PaymentType_must_be_cash_or_card check(PaymentType in ('Cash', 'Card')),
+    constraint ck_Orders_DateOrdered_must_be_before_DateNeeded check(DateOrdered < DateNeeded),
```
Comment (SOsdoba) [inline BakeryDB/1 tables.sql]: I don't think you need this constraint, as the following constraint already takes care of the DateOrdered being before the DateNeeded.

File: BakeryDB/1 tables.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    OrdersId int not null identity primary key,
+    FirstName varchar(30) not null constraint ck_Orders_FirstName_cannot_be_blank check(FirstName <> ''),
+    LastName varchar(30) not null constraint ck_Orders_LastName_cannot_be_blank check(LastName <> ''),
+    Item varchar(30) not null constraint ck_Orders_Item_cannot_be_blank check(Item <> ''),
+    Quantity int not null constraint ck_Orders_Quantity_must_be_greater_than_zero check(Quantity >= 0),
+    PricePerUnit decimal(6,2) not null constraint ck_Orders_PricePerUnit_must_be_between_2_dollars_and_50_cents_and_650_dollars check(PricePerUnit between 2.5 and 650),
+    TotalCost as Quantity * PricePerUnit,
+    DateOrdered date not null constraint ck_DateOrdered_must_be_before_the_current_date check(DateOrdered <= getdate()),
+    DateNeeded date not null constraint ck_Orders_DateNeeded_must_be_after_January_1_2023 check(DateNeeded >= '2023-01-01'),
+    CreditCardNum char(4) not null constraint ck_Orders_CreditCardNum_must_be_numbers check(CreditCardNum like ('[0-9][0-9][0-9][0-9]') or CreditCardNum = ''),
```
Comment (SOsdoba) [inline BakeryDB/1 tables.sql]: There should be a constraint that ensures that CreditCardNum only allows blank when the payment method is cash.

File: BakeryDB/1 tables.sql [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+drop table if exists Orders
+go
+
+create table dbo.Orders(
+    OrdersId int not null identity primary key,
+    FirstName varchar(30) not null constraint ck_Orders_FirstName_cannot_be_blank check(FirstName <> ''),
+    LastName varchar(30) not null constraint ck_Orders_LastName_cannot_be_blank check(LastName <> ''),
+    Item varchar(30) not null constraint ck_Orders_Item_cannot_be_blank check(Item <> ''),
+    Quantity int not null constraint ck_Orders_Quantity_must_be_greater_than_zero check(Quantity >= 0),
+    PricePerUnit decimal(6,2) not null constraint ck_Orders_PricePerUnit_must_be_between_2_dollars_and_50_cents_and_650_dollars check(PricePerUnit between 2.5 and 650),
```
Comment (SOsdoba) [inline BakeryDB/1 tables.sql]: I don't think it's necessary to allow for 4 digits on the left side of the decimal as the most we will charge is $650, which would only make it necessary for 3 digits on the left side of the decimal.

File: BakeryDB/3 reports.sql [hunk 4]
```diff
@@ -0,0 +1,46 @@
+--1. I will need a list of my top 3 goods that are the most popular.
+select top 3 NumberOfOrders = count(*), o.Item
+from orders o
+group by o.Item
+order by NumberOfOrders desc
+
+/*2. I would like to know how many customers are paying with cash and how many with card, just so I can know how to better service them. 
+  (Eventually I would like to switch over to card only.). 
+  I also need to see the average spent per payment method.*/
+select count(*), AvgSpent = avg(o.TotalCost), o.PaymentType
```
Comment (SOsdoba) [inline BakeryDB/3 reports.sql]: The first column needs a name.

File: BakeryDB/3 reports.sql [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+from Orders o
+order by MonthOrdered desc
+
+/*4. People often complain that my prices are exorbatant. 
+  I was thinking to have a "Buy 1 get 1 free" for the first and 15th of each month. 
+  I would like to know the total loss, as well as the loss per month.*/
+select TotalLoss = sum(o.TotalCost) - sum(case when datepart(day, o.DateOrdered) in(1,15) and o.Quantity > 1 then (o.Quantity / 2) * o.PricePerUnit else o.TotalCost end)
+from orders o
+
+select LossPerMonth = sum(o.TotalCost) - sum(case when datepart(day, o.DateOrdered) in(1,15) and o.Quantity > 1 then (o.Quantity / 2) * o.PricePerUnit else o.TotalCost end), datename(month, o.DateOrdered)
```
Comment (SOsdoba) [inline BakeryDB/3 reports.sql]: The column with the months need a name.

**Review by SOsdoba — CHANGES_REQUESTED** [review]
I added some comments to the table and reports files.

File: BakeryDB/1 tables.sql (hunk 1 above)
Comment (AvigailGellis) [reply BakeryDB/1 tables.sql]: Fixed.

File: BakeryDB/1 tables.sql (hunk 2 above)
Comment (AvigailGellis) [reply BakeryDB/1 tables.sql]: Fixed.

File: BakeryDB/1 tables.sql (hunk 3 above)
Comment (AvigailGellis) [reply BakeryDB/1 tables.sql]: Fixed.

File: BakeryDB/3 reports.sql (hunk 4 above)
Comment (AvigailGellis) [reply BakeryDB/3 reports.sql]: Fixed.

File: BakeryDB/3 reports.sql (hunk 5 above)
Comment (AvigailGellis) [reply BakeryDB/3 reports.sql]: Fixed.

File: BakeryDB/1 tables.sql (hunk 3 above)
Comment (SOsdoba) [reply BakeryDB/1 tables.sql]: Great

File: BakeryDB/1 tables.sql (hunk 2 above)
Comment (SOsdoba) [reply BakeryDB/1 tables.sql]: Great

File: BakeryDB/3 reports.sql (hunk 4 above)
Comment (SOsdoba) [reply BakeryDB/3 reports.sql]: Great

File: BakeryDB/3 reports.sql (hunk 5 above)
Comment (SOsdoba) [reply BakeryDB/3 reports.sql]: Great

File: BakeryDB/1 tables.sql (hunk 1 above)
Comment (SOsdoba) [reply BakeryDB/1 tables.sql]: great

**Pull request closed without merging**